### ISSUE TITLE
Cognito doesn't support refresh token rotation

### ISSUE THREAD

**Issue by kainanftf** [AxaGuilDEv/react-oidc#847]
Hello and thanks for the lib! 

Currently trying to use this lib with Cognito however running into some issues when refreshing tokens. It seems Cognito does not use refresh token rotation and intends for the refresh token to be re-used until expired. This behavior is consistent with their [docs](https://docs.aws.amazon.com/cognito/latest/developerguide/token-endpoint.html) and is mentioned in this Amplify(sdk that works with Cognito)[ issue](https://github.com/aws-amplify/amplify-js/issues/5214).

Basically since the response from the `/token` endpoint only includes access/id tokens the lib seems to be setting the response as the new tokens causing the SPA to lose the refresh token. Here is a quick video showing what I mean:

https://user-images.githubusercontent.com/85503587/187739961-91814b6d-a1b9-4177-b578-787a8a6b97d5.mp4

Was trying to dig into the lib to see where this is handled and maybe patch, but got a bit lost...

**Comment by guillaume-chervet** [comment]
Hi @kainanftf, thank you very much for your issue and your awesome video. I am going to sleep now. 
I will try to add this behavior tomorrow. To keep the first refresh token when /token endpoint is called and no refreshToken is found inside the response.

I have used cognito in the past and I like it :)

I have to clean the source code to make it more accessible.
Oidc protocol is simple and complexe at the same time ^^

**Comment by kainanftf** [comment]
@guillaume-chervet 
Thanks so much, I hope you get some good rest...Really appreciate the quick response as well!

> I have to clean the source code to make it more accessible.
Oidc protocol is simple and complexe at the same time ^^

No worries, just happy to be able to leverage the library ^^

**Comment by guillaume-chervet** [comment]
Hi @kainanftf , I have published a fix. It may work but I cannot test because I do not have any cognito for the moment.


Does it work for you? 
Thank you very mucv for your help that will make this library better :)

**Comment by kainanftf** [comment]
@guillaume-chervet 
Hey, can confirm that it is indeed working! Thanks so much, very appreciated 😄 .
